aoc/src/main/scala/aoc2015/Day04.scala

```diff
@@ -0,0 +1,22 @@
+package aoc2015
+
+object Day04 extends App:
+
+  import java.security.MessageDigest
+
+  def hash(s: String) = {
+    val m = java.security.MessageDigest.getInstance("MD5")
+    val b = s.getBytes("UTF-8")
+    m.update(b, 0, b.length)
+    new java.math.BigInteger(1, m.digest())
```

Comment:
works - but would prefer a diy converter that doesn't require the array to be traversed trice, eg.

```
def convertBytesToHex(bytes: Array[Byte]): String =
  val sb = new StringBuilder
  for (b <- bytes)(sb.append(String.format("%02x", Byte.box(b))))
  sb.toString
```